cmd/snap-update-ns/change.go

```diff
@@ -59,26 +60,145 @@ var changePerform func(*Change) ([]*Change, error)
 
 // changePerformImpl is the real implementation of Change.Perform
 func changePerformImpl(c *Change) ([]*Change, error) {
+	var changes []*Change
 	if c.Action == Mount {
+		// We may be asked to bind mount a file, bind mount a directory, mount
+		// a filesystem over a directory or create a symlink (which is abusing
+		// the "mount" concept slightly). That actual operation is performed in
+		// c.lowLevelPerform. Here we just set the stage to make that possible.
+		kind, _ := c.Entry.OptStr("x-snapd.kind")
+		path := c.Entry.Dir
+
+		// We use lstat to ensure that we don't follow a symlink in case one
+		// was set up by the snap. Note that at the time this is run, all the
+		// snap's processes are frozen.
+		fi, err := osLstat(path)
+
+		// In case we need to create something, some constants.
 		const (
 			mode = 0755
 			uid  = 0
 			gid  = 0
 		)
-		// Create target mount directory if needed.
-		if err := ensureMountPoint(c.Entry.Dir, mode, uid, gid); err != nil {
-			return nil, err
+
+		// tryIt takes one flag argument
+		type tryItFlag int
+		const createSymlinks tryItFlag = 1
+
+		tryIt := func(flags tryItFlag) error {
```

Comment:
maybe I'm having a bad day, but defining the helper like this seems to me to be a poor choice for readability (when compared to defining it at the same level as `changePerformImpl`). Granted then you'd have to solve giving a proper name :-)

Reply:
I agree with @chipaca , the helper is OK, but we will benefit from it more when it becomes a function on its own at the top level. In that case tryIt is not the best name ;), how about something like `secureMk` or `secureMkImpl` (still not perfect, but more descriptive)?

Reply:
When it's a separate function I have to pass all of the of arguments it currently gets as context, I really liked how it was before (simple) even at the cost or some repetition.

Can we agree on a better name but keep it internal to this function?

Reply:
Ah, right, fair enough, no need to overkill that.